nifi-nar-bundles/nifi-standard-services/nifi-hbase_1_1_2-client-service-bundle/nifi-hbase_1_1_2-client-service/src/main/java/org/apache/nifi/hbase/HBase_1_1_2_ClientService.java

```diff
@@ -549,11 +551,7 @@ public String toTransitUri(String tableName, String rowKey) {
             logger.warn("Connection has not been established, could not create a transit URI. Returning null.");
             return null;
         }
-        try {
-            final String masterAddress = connection.getAdmin().getClusterStatus().getMaster().getHostAndPort();
-            return "hbase://" + masterAddress + "/" + tableName + (rowKey != null && !rowKey.isEmpty() ? "/" + rowKey : "");
-        } catch (IOException e) {
-            throw new RuntimeException("Failed to get HBase Admin interface, due to " + e, e);
-        }
+        final String transitUriMasterAddress = StringUtils.isEmpty(masterAddress) ? "unknown" : masterAddress;
```

Comment:
We are sending back "unknown" if the master address is empty. Won't that be a problem, if it happens to be empty? I mean when `session.getProvenanceReporter().fetch(handlerFlowFile, transitUri);` gets executed on the HBase processors?

Reply:
@zenfenan Thanks for your comment. Yes, it would be a problem if this actually happens. But it only happens when the ControllerService was not able to get the master address when it was enabled. I think HBase processors will not be able to access HBase, too.

It depends on the programs how critical this can be, for those use the returned transit URI later. For example, NiFi and Atlas integration case, ReportLineageToAtlas reporting task will not be able to determine the right cluster name in Atlas.

Please let me know any situation that the ControllerServer can connect to a HBase, but can not get its master address. I write the code that way just to be more protective, but not sure if that actually happens.

Reply:
As far as I'm aware, such a scenario has no possibilities of happening. I just wanted to start the thinking in that direction so that people who know in and out of HBase can pitch in their thoughts. 

It looks good to me. Thanks :)

Reply:
@ijokarumawak I think we might need to dig into the HBase API and see if there is a way we can watch for a leader election w/ the master.

Reply:
@ijokarumawak I think a timer that asks the cluster for the current master on a configurable interval would be good to have here. It could have a default of 5-15 minutes. I skimmed through the Java API for HBase and didn't see anything that we can use for that, but we should periodically make sure the current master info is correct so the provenance info is correct(ish).

Reply:
@MikeThomsen That would make provenance info more accurate. I think that feature will be helpful when the HBase master is failed over.

Again, it depends on the other program those uses the provenance event how it is important. For the particular use-case of NiFi and Atlas integration, it is sufficient to know which HBase 'cluster' NiFi is talking to. If there are multiple master addresses used for HA purpose, then any of those would be enough to identify a cluster. E.g. cluster A (master1 and master2) and cluster B (master3 and 4), then it's not significant if it is master1 or 2 to tell NiFi is accessing cluster A.

I agree with your suggestion to make this more accurate, and good to have. But I'd prefer to do that separately. Would you create another JIRA if you don't mind? Thank you!

Reply:
Created the jira and linked it to this one PR's ticket.